### DISCUSSION TITLE
QM Caching Question

### DISCUSSION THREAD

**Discussion by mikemhenry** [openforcefield/openff-bespokefit#309] [Q&A] [answered]
How does the local QM data cache work for a production fit? Like if the number of steps in the torsion drive change or a different level of theory is used, will it create new jobs or if there was a fragment that had a higher number of steps than the new job you submitted, will it use that data or generate more data?

**Comment by jthorton** [comment, marked as answer]
Hi @mikemhenry sorry for the delay, the cache is based on the [Torsion1DTask](https://github.com/openforcefield/openff-bespokefit/blob/e8859123afd439d2164badfaf1404ccc838b3a0a/openff/bespokefit/schema/tasks.py#L77) schema. Two schemas must be identical for the cache to be triggered (this covers program, method, basis, optimisation and torsiondrive settings) as well as the molecules and the targeted torsion. We try to ensure that the molecules are always in [canonical order](https://github.com/openforcefield/openff-bespokefit/blob/e8859123afd439d2164badfaf1404ccc838b3a0a/openff/bespokefit/executor/services/qcgenerator/cache.py#L14) when creating the task and that the central atoms of the torsion are tagged as atom 1 and 2 to maximise the hit rate.
